### ISSUE TITLE
Implement wrapping for the tab line

### ISSUE THREAD

**Issue by osa1** [osa1/tiny#125]
Currently if I have more tabs than what fits on the screen we draw left and right arrows and implement scrolling. It looks like this:

![screenshot_2019-02-20_08-59-22](https://user-images.githubusercontent.com/448274/53069666-d6f84880-34ed-11e9-8e65-340880c6da81.png)

I implemented this but I never really needed it as I wasn't joining too many channels/servers. But now I do, and I realize that while this look OK it's really hard to use:

- I don't see which channels have updates without scrolling. The arrows change colors as channels that are now shown get updates, but that's not good enough as it still doesn't tell me which channels have updates.
- I don't see underlined characters in the channels that are not shown on the screen, so no way to jump quickly to those channels.

So I think it'd be better to render channels in multiple lines when they don't fit in one line.

No need to remove the current behavior -- we can put it behind a setting in the config file. Also, we may want to use the current behavior when the screen is too small to allocate multiple lines to the channel list.

**Comment by trevarj** [comment]
I was hoping that this would be my next task, but after some tinkering I got stuck at a design decision. 

Allowing the tabs to wrap is actually really simple, and I have already built a `TabWidget` for the `TUI` to contain all the relevant code for this in its own space.

Personally, I don't like the horizontal scrolling at all, even for fallback. I think it would be nice to rip out all of the horizontal scrolling logic and convert it to vertical scrolling, which would only be used when the screen is small and the tab line is a certain size (maybe greater than 1/6th of the total screen height?).  I think that a vertical scroll would be simpler to implement, but the negative side is that you don't see as many tabs in scroll mode.

I also would like to add a control to hide/show the tab line, which could also be useful for when the screen is small. This control could also function as an "expand" to make the tab line full size when it is in scroll mode.

**Vertical scroll idea:**

Full view (or "expanded" tab line mode):
```
|                      |
|                      |
|osa1:                 |
|mentions #line1       |
|#random #line2 #chan  |
|#line3 #mychan        |
```
Top line (small screen):
```
|                      |
|                      |
|osa1:                 |
|#random #line2 #chan ↕|
```
Middle line (small screen):
```
|                      |
|                      |
|osa1:                 |
|mentions #line1      ↓|
```
Bottom line (small screen):
```
|                      |
|                      |
|osa1:                 |
|#line3 #mychan       ↑|
```
